### PR TITLE
jfrog-cli: 2.52.10 -> 2.55.0

### PR DESCRIPTION
**Pull request by aidalgol**
## Description of changes
* Set update script.
* Update to latest release.
* Add myself as maintainer.

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Detegr — APPROVED** [review]
LGTM. Sorry for taking this long to review.

**Comment by wegank** [comment]
@ofborg build jfrog-cli jfrog-cli.passthru.tests

File: pkgs/tools/misc/jfrog-cli/default.nix [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
   src = fetchFromGitHub {
     owner = "jfrog";
     repo = "jfrog-cli";
     rev = "refs/tags/v${version}";
-    hash = "sha256-sqKlYHOpjIxNff1QK540Xxqr7w+WZ+pZXgfAKjRbMuM=";
+    hash = "sha256-GwhEkoPHBsWyEO0H7BU7HM6t8aJ6iNKktkYiyFj9c80=";
   };
 
   vendorHash = "sha256-Cm6Wu2U2i6WZfps1WLAjYuVZZ1y7J4WXFLmX2bkevd4=";
```
Comment (wegank) [inline pkgs/tools/misc/jfrog-cli/default.nix:18]: 4 neutral checks, doesn't look good. Does this need to be updated as well?

Comment (aidalgol) [reply pkgs/tools/misc/jfrog-cli/default.nix]: You're right, it does.  I should have used `nix-update-script` for `passthru.updateScript`.  I'm sorry, I don't know how I didn't catch this.

**Review by Detegr — APPROVED** [review]

**Comment by wegank** [comment]
Still 4 neutral checks.

**Comment by aidalgol** [comment]
I'm not sure what's going on now.  The update script updated the `vendorHash`, but ofborg still complains with a hash mismatch.

**Comment by aidalgol** [comment]
I think I've fixed it.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 301375` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jfrog-cli</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**